### ISSUE TITLE
Pyreverse duplicate arrows

### ISSUE THREAD

**Issue by nickdrozd** [pylint-dev/pylint#8522]
### Bug description

Similar to https://github.com/pylint-dev/pylint/issues/8189, field type declarations and assignments are recorded separately, resulting in duplicate arrows.

Example:

```python
class A:
    pass

class B:
    a: A

    def __init__(self):
        self.a = A()
```

![classes](https://user-images.githubusercontent.com/17630138/229254737-d769d9ee-7c97-4a40-98eb-d8ad34bc9163.png)

### Command used

```shell
pyreverse --only-classnames -o png asdf.py
```

### Pylint output

```shell
...
```

### Expected behavior

Just one arrow per field.

### Pylint version

```shell
pylint 3.0.0a6
astroid 2.16.0dev0
Python 3.11.0 (tags/v3.11.0:deaf509e8f, Nov  9 2022, 13:21:28) [GCC 9.4.0]
```

**Comment by mbyrnepr2** [comment]
The example looks logically identical to the one in 8189; are they different issues?

**Comment by DudeNr33** [comment]
I thought so too at first. 
It's probably a special case of this issue and may very well have the same root cause as #8189. 
It would be best to first try to tackle the referenced issue and then check if this problem here is solved as well.
